DEVDOCS-4641: [new] Tax Provider and Connection API, added profile option (#1728)
bigcommerce/dev-docs: docs/api-docs/provider-apis/tax/overview.mdx

The [Tax Provider API](/docs/apps-api/tax) works in conjunction with a BigCommerce app, so you will need to [build an app](/api-docs/apps/guide/intro) that integrates the [Tax Provider API](/docs/apps-api/tax).

Multi-tenant tax providers can choose to publish their BigCommerce app so that it's discoverable by anyone, or publish their app as unlisted so that it can only be installed using a URL. Furthermore, tax solutions that have been built in-house or for specific merchants by digital agencies are also supported as private instances, and will only work on the specified stores.

## Obtaining an app ID

To get your app ID, start creating an app in the [Developer Portal](https://devtools.bigcommerce.com/), and fill out the information on [Step 2 Technical](/api-docs/apps/guide/publishing#add-technical-information). In the URL, you will see your unique app ID.

[Learn more about finding your app's ID](/api-docs/apps/tutorials/id).

## Sharing provider details with BigCommerce

> Once you've obtained your app ID, share your app ID as well as the below information with BigCommerce by emailing [taxproviderapi@bigcommerce.com](mailto:taxproviderapi@bigcommerce.com). This allows us to create your tax provider configuration, which will take approximately five days.

Reviewer: ```suggestion
Once you've obtained your app ID, share your app ID and the information below with BigCommerce by emailing [taxproviderapi@bigcommerce.com](mailto:taxproviderapi@bigcommerce.com). This information allows us to create your tax provider configuration, which will take approximately five days.
```

> Once your tax provider configuration is ready, we'll let you know through email. The email will also include your `provider_id` which is required when [establishing a connection](#establishing-a-connection) with the [Tax Provider API](/docs/apps-api/tax).

Reviewer: ```suggestion
Once your tax provider configuration is ready, we'll let you know through email. The email will also include your `provider_id`, which is required when [establishing a connection](#establishing-a-connection) with the [Tax Provider API](/docs/apps-api/tax).
```

| Tax provider name | Required | String | Displayed in the active MSF-enabled BigCommerce control panel (e.g. **Settings > Setup > Tax > Add tax service**). | `Sample Tax` |
| Tax provider type | Required | Production, Sandbox | Hierarchy of tax provider configurations, Production is primary and Sandboxes are secondary, see [sandbox tax provider configuration](#sandbox-tax-provider-configuration) for more information. | `Production` |
| Partner support email | Required | Email | Used by BigCommerce to contact tax provider, in the case we need to forward a merchant support request. | `support@sampletax.example.com` |

> | Links displayed in an active MSF-enabled control panel | Optional | URL, Title, Description | Link(s) displayed to merchants when they navigate to **Settings > Setup > Tax > _TaxProviderName_**. One or multiple links supported. | `support.sampletax.com`, `Sample Tax Support`, `The Sample Tax Support website.` |

Author: Minor question, should this be changed to merchants or **both** merchants and partners? (i.e. do partner stores have the same UI as merchant stores?)

Reviewer 2: I think merchants is fine, as when a partner is using a store they are acting as a merchant. I.e. the fact it is a partner is irrelevant tot he site itself.

**Coverage** | | | | |

> | Tax provider visibility | Required | Show, Hide | Tells us if tax provider should be displayed to merchants in the BigCommerce control panel before the tax providers app is installed on the associated store. | `Show` |

Author: Minor question, are the users merchants or **both** merchants and partners?

| <sup>+</sup>Void URL | Optional | URL | URL BigCommerce should use for Tax Provider API quote requests. | `https://sampletax.example.com/doc/void` 
| **Testing** | | | | |
| Partner sandbox store domain | Required | Domain name | Share your partner sandbox store for testing purposes prior to launching your tax provider. Learn how to [create a partner sandbox store](/api-docs/partner/getting-started/create-a-sandbox-store). | `https://sampletax-test-store.mybigcommerce.com/` |

<sup>+</sup> You have the option of supporting a URL that merchants can customize. See [Tax Profile](#tax-profile-optional).

### Tax profile (optional)

BigCommerce sets the tax provider registry with your URLs for [Tax Provider API](/docs/apps-api/tax) requests. BigCommerce sends requests to these URLs, for example, when a shopper requests tax estimates.
If you would like to support multiple connections for a merchant, BigCommerce can set the registry with flexible URLs that use our new profile option. 
These URLs would include a reserved string, for example, `https://{profile}/estimate`, allowing merchants to customize the base domain.   

> You will need to allow merchants to enter the URL's base domain in the control panel when they connect or install your app. Merchants would also need to be able to self-serve updates to the URL through your app. 

Reviewer: ```suggestion
When connecting or installing your app, you must allow merchants to enter the URL's base domain in the control panel. Merchants need to be able to self-serve updates to the URL through your app. 
```

> When you [Establish a Tax Provider connection](#establishing-a-connection), you will include the `profile` base domain along with credentials associated with the merchant's account. 

Reviewer 3: ```suggestion
When you [Establish a Tax Provider connection](#establishing-a-connection), include the `profile` base domain along with credentials associated with the merchant's account. 
```

<Callout type="info">

>   You must liaise with the merchant to provide them with customized URLs. BigCommerce merely consumes the URL values that the merchant enters and uses the provided string value for Tax Provider API requests.

Reviewer: ```suggestion
  You must liaise with the merchant to provide them with customized URLs. BigCommerce merely consumes the URL values the merchant enters and uses the provided string value for Tax Provider API requests.
```

## Building the app

> Tax providers are required to build a BigCommerce [single-click app](/api-docs/apps/guide/types#single-click) in order to utilise the Tax Provider API to provide tax estimates and submit tax documents. A BigCommerce single-click app provides many benefits, for example, it enables tax providers to promote their solution in the BigCommerce App Marketplace, ask for merchant authorisation of API scopes during app install, as well as enable configuration of tax provider settings through an iFrame in the BigCommerce control panel.

Reviewer: ```suggestion
Tax providers must build a BigCommerce [single-click app](/api-docs/apps/guide/types#single-click) to utilize the Tax Provider API to provide tax estimates and submit tax documents. A BigCommerce single-click app provides many benefits. For example, it enables tax providers to promote their solution in the BigCommerce App Marketplace, ask for merchant authorization of API scopes during app install, and allow configuration of tax provider settings through an iFrame in the BigCommerce control panel.
```

## Enabling tax providers in the control panel

> Once the tax provider's app is successfully installed and basic authentication credentials are provided through the Update a Connection request, merchants can enable the tax provider on all test stores provided by the tax provider in [sharing provider details with BigCommerce](#sharing-provider-details-with-bigcommerce).

Reviewer: ```suggestion
Once you have successfully installed the tax provider's app and provided basic authentication credentials through the Update a Connection request, merchants can enable the tax provider on all test stores supplied by the tax provider in [sharing provider details with BigCommerce](#sharing-provider-details-with-bigcommerce).
```

> Once the tax provider's app has been successfully installed and basic authentication credentials have been provided via the Update a Connection request, merchants will be able to enable the tax provider on all test stores provided by the tax provider in [sharing provider details with BigCommerce](#sharing-provider-details-with-bigcommerce).

Reviewer 3: ```suggestion
Once the tax provider's app is successfully installed and basic authentication credentials are provided through the Update a Connection request, merchants can enable the tax provider on all test stores provided by the tax provider in [sharing provider details with BigCommerce](#sharing-provider-details-with-bigcommerce).
```

> To enable the tax provider, merchants must navigate to **Settings > Setup > Tax** in an active MSF-enabled BigCommerce control panel and click **Enable** next to the associated tax provider.

Author: Minor question, should this be changed to merchants or **both** merchants and partners? (i.e. do partner stores have the same UI as merchant stores?)

Author: Got my answer [here](https://github.com/bigcommerce/dev-docs/pull/1728#discussion_r1152568576)

> Estimate calls will also be made in the control panel when using flows like the following.

Reviewer: ```suggestion
Estimate calls will also be made in the control panel when using flows like the following:
```

* Line-item refund flows

> * Test connection functionality when merchants navigate to **Settings > Setup > Tax > _TaxProviderName_** in an active MSF-enabled BigCommerce control panel

Author: Minor question, should this be changed to merchants or **both** merchants and partners? (i.e. do partner stores have the same UI as merchant stores?)

Author: Got my answer [here](https://github.com/bigcommerce/dev-docs/pull/1728#discussion_r1152568576)

* The tax provider, found by navigating to **Settings > Setup > Tax**, is enabled
* If document submission is supported, navigate to **Settings > Setup > Tax > _TaxProviderName_** and ensure the **Submit Order Data** checkbox is checked

> To test the tax provider connection, navigate to **Settings > Setup > Tax > _TaxProviderName_** in an active MSF-enabled BigCommerce control panel and click **Test Connection**. This will trigger a sample tax estimate request to be sent to the estimate URL provided by the tax provider. If the connection is unsuccessful, merchants can navigate to **Settings > Advanced > Store logs** in an active MSF-enabled BigCommerce control panel to view the error and its description to assist with triaging the issue.

Author: Minor question, should this be changed to merchants or **both** merchants and partners? (i.e. do partner stores have the same UI as merchant stores?)

Author: Got my answer [here](https://github.com/bigcommerce/dev-docs/pull/1728#discussion_r1152568576)

> To test the tax provider connection, navigate to **Settings > Setup > Tax > _TaxProviderName_** in an active MSF-enabled BigCommerce control panel and click **Test Connection**. This will trigger a sample tax estimate request to be sent to the estimate URL provided by the tax provider. If the connection is unsuccessful, merchants can navigate to **Settings > Advanced > Store logs** in an active MSF-enabled BigCommerce control panel to view the error and its description to assist with triaging the issue.

Reviewer: ```suggestion
To test the tax provider connection, navigate to **Settings > Setup > Tax > _TaxProviderName_** in an active MSF-enabled BigCommerce control panel and click **Test Connection**. This selection will trigger a sample tax estimate request to be sent to the estimate URL provided by the tax provider. If the connection is unsuccessful, merchants can navigate to **Settings > Advanced > Store logs** in an active MSF-enabled BigCommerce control panel to view the error and its description to assist with triaging the issue.
```
